scene/animation/animation_player.compat.inc

```diff
@@ -0,0 +1,76 @@
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef DISABLE_DEPRECATED
+
+void AnimationPlayer::_set_process_callback_bind_compat_80813(AnimationPlayer::AnimationProcessCallback p_mode) {
+	set_callback_mode_process(static_cast<AnimationManager::AnimationCallbackModeProcess>(static_cast<int>(p_mode)));
+}
+
+AnimationPlayer::AnimationProcessCallback AnimationPlayer::_get_process_callback_bind_compat_80813() const {
+	return static_cast<AnimationProcessCallback>(static_cast<int>(get_callback_mode_process()));
+}
+
+void AnimationPlayer::_set_method_call_mode_bind_compat_80813(AnimationPlayer::AnimationMethodCallMode p_mode) {
+	set_callback_mode_method(static_cast<AnimationManager::AnimationCallbackModeMethod>(static_cast<int>(p_mode)));
+}
+
+AnimationPlayer::AnimationMethodCallMode AnimationPlayer::_get_method_call_mode_bind_compat_80813() const {
+	return static_cast<AnimationMethodCallMode>(static_cast<int>(get_callback_mode_method()));
+}
+
+void AnimationPlayer::_set_root_bind_compat_80813(const NodePath &p_root) {
+	set_root_node(p_root);
+}
+
+NodePath AnimationPlayer::_get_root_bind_compat_80813() const {
+	return get_root_node();
+}
+
+void AnimationPlayer::_seek_bind_compat_80813(double p_time, bool p_update) {
+	seek(p_time, p_update, false);
+}
+
+void AnimationPlayer::_bind_compatibility_methods() {
+	ClassDB::bind_method(D_METHOD("set_process_callback", "mode"), &AnimationPlayer::_set_process_callback_bind_compat_80813);
+	ClassDB::bind_method(D_METHOD("get_process_callback"), &AnimationPlayer::_get_process_callback_bind_compat_80813);
+	ClassDB::bind_method(D_METHOD("set_method_call_mode", "mode"), &AnimationPlayer::_set_method_call_mode_bind_compat_80813);
+	ClassDB::bind_method(D_METHOD("get_method_call_mode"), &AnimationPlayer::_get_method_call_mode_bind_compat_80813);
+	ClassDB::bind_method(D_METHOD("set_root", "path"), &AnimationPlayer::_set_root_bind_compat_80813);
+	ClassDB::bind_method(D_METHOD("get_root"), &AnimationPlayer::_get_root_bind_compat_80813);
+	ClassDB::bind_compatibility_method(D_METHOD("seek", "seconds", "update"), &AnimationPlayer::_seek_bind_compat_80813, DEFVAL(false));
+	BIND_ENUM_CONSTANT(ANIMATION_PROCESS_PHYSICS);
+	BIND_ENUM_CONSTANT(ANIMATION_PROCESS_IDLE);
+	BIND_ENUM_CONSTANT(ANIMATION_PROCESS_MANUAL);
+	BIND_ENUM_CONSTANT(ANIMATION_METHOD_CALL_DEFERRED);
+	BIND_ENUM_CONSTANT(ANIMATION_METHOD_CALL_IMMEDIATE);
```

Comment:
I tested commenting out the BIND_ENUM_CONSTANT and it seems to avoid adding the unneeded constant documentation to AnimationPlayer and AnimationTree at least in godot-rust

Reply:
Though... technically speaking, a GDScript user might have referenced `AnimationPlayer.ANIMATION_PROCESS_IDLE` in Godot 4.1, and so having these enum constants would help keep compatibility with GDScript code.

So maybe best not to remove the exposed enum constants for script compat.

Reply:
Well, I will just leave it there for now. Thank you for the testing.